yt_dlp/extractor/bilibili.py

```diff
@@ -774,3 +775,139 @@ def _real_extract(self, url):
+                    'vbr': video_res.get('bandwidth'),
+                    'acodec': 'none',
+                    'vcodec': video_res.get('codecs'),
+                    'filesize': video_res.get('size'),
+                })
+        for aud in video_json.get('audio_resource', []):
+            if aud.get('url'):
+                formats.append({
+                    'url': aud['url'],
+                    'ext': 'mp4',
+                    'abr': aud.get('bandwidth'),
+                    'acodec': aud.get('codecs'),
+                    'vcodec': 'none',
+                    'filesize': aud.get('size'),
+                })
+        sub_json = self._call_api(f'/m/subtitle?ep_id={ep_id}&platform=web', ep_id)
+        subtitles = {}
+        for sub in sub_json.get('subtitles', []):
+            sub_url = sub.get('url')
+            if not sub_url:
+                continue
+            subtitles.setdefault(sub.get('key', 'en'), []).append({
+                'url': sub_url,
+            })
+        self._sort_formats(formats)
+        return formats, subtitles
+
+    def _extract_ep_info(self, episode_data, ep_id):
+        formats, subtitles = self._extract_formats_and_subtitles(ep_id)
+        return {
+            'id': ep_id,
+            'title': episode_data['long_title'] or episode_data.get('title'),
+            'thumbnail': episode_data.get('cover'),
+            'episode_number': str_to_int(episode_data.get('title')),
+            'formats': formats,
+            'subtitles': subtitles,
+        }
+
+
+class BiliIntlIE(BiliIntlBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?bili(?P<type>bili\.tv|intl.com)/(?:[a-z]{2}/)?play/(?P<season_id>\d+)/(?P<id>\d+)'
+    _TESTS = [{
+        'url': 'https://www.bilibili.tv/en/play/34613/341736',
+        'info_dict': {
+            'id': '341736',
+            'ext': 'mp4',
+            'title': 'The First Night',
+            'thumbnail': 'https://i0.hdslb.com/bfs/intl/management/91e30e5521235d9b163339a26a0b030ebda54310.png',
+            'episode_number': 2,
+        },
+        'params': {
+            'format': 'bv',
+        },
+    }, {
+        'url': 'https://www.biliintl.com/en/play/34613/341736',
+        'info_dict': {
+            'id': '341736',
+            'ext': 'mp4',
+            'title': 'The First Night',
+            'thumbnail': 'https://i0.hdslb.com/bfs/intl/management/91e30e5521235d9b163339a26a0b030ebda54310.png',
+            'episode_number': 2,
+        },
+        'params': {
+            'format': 'bv',
+        },
+    }]
+
+    def _real_extract(self, url):
+        type, season_id, id = self._match_valid_url(url).groups()
+        self._API_URL = self._API_URL.format(type)
+        data_json = self._call_api(f'/web/view/ogv_collection?season_id={season_id}', id)
+        for episode in data_json.get('episodes', []):
+            if str(episode.get('ep_id')) == id:
+                episode_data = episode
+                break
```

Comment:
sacrificing time here in favor of saving a ton in series extractor.

Reply:
Whenever you are see the pattern 
```py
for x in y:
    if f(x):
        correct_x = x
        break
```

it can be changed to `correct_x = next(x for x in y if f(x))` or `correct_x = next(filter(f, y))`

```suggestion
        episode_data = next(
            episode for episode in data_json.get('episodes', [])
            if str(episode.get('ep_id')) == id)
```